### PR TITLE
Allow users to update their own password

### PR DESCRIPTION
**Pull request by pawiecz**
This PR is an initial implementation of a feature requested as #298. It is still missing method in the `kernelci.api` base class, command for `kci user` as well as proper unit and e2e tests.

Fixes https://github.com/kernelci/kernelci-api/issues/298

### CODE REVIEW

**Comment by pawiecz** [comment]
I'm definitely missing a key part to run e2e tests because mine is continuously getting skipped:

1. Dependency donor is marked with `@pytest.mark.dependency()`
2. Dependency donor passes successfully
3. Dependency taker uses the same naming as pytest (`e2e_tests/test_user_creation.py::test_create_regular_user`)

Yet my test is not executed at all

**Comment by pawiecz** [comment]
Patches were squashed and force-pushed because `fixup!` commit didn't pass DCO check.

**Comment by pawiecz** [comment]
More fixups will probably come with the review, I'll postpone squashing them until the comment threads are resolved.

**Comment by pawiecz** [comment]
Here's a list of tests this (or next) PR should probably include to cover this feature better:

- nonexistent user
- wrong password for existing user
- empty new password (handler should probably reject such requests from making any changes)

**Comment by JenySadadia** [comment]
The test looks good to me.
Although we are not running `pylint` for e2e tests atm, it would be nice to fix some of the pylint errors for this PR if we can.
You can check the annotated inline errors here https://github.com/kernelci/kernelci-api/pull/320/files.

File: api/main.py [hunk 1]
```diff
@@ -352,6 +352,29 @@ async def whoami(current_user: User = Depends(get_user)):
...
+async def reset_password(
+        username: str, current_password: Password, new_password: Password):
+    """Set a new password for an authenticated user"""
+    authenticated = await auth.authenticate_user(
+        username, current_password.password.get_secret_value())
+    if not authenticated:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
+        )
```
[the remark this answers is not in the document]
Comment (pawiecz) [reply api/main.py:365]: I'll add myself a TODO for this. Could that injection be also reused in the [`/token` endpoint](https://github.com/kernelci/kernelci-api/blob/8a9572750edaecbfdcca2835a36776647adde8ca/api/main.py#L319) (that's where I first saw this flow)?

**Comment by pawiecz** [comment]
@JenySadadia Thanks for the pointer to the linter annotations! I reformatted new e2e tests file with a `black` command. I also dropped:
- explicit dependency marker (pytest deps work fine without it)
- intermediary var for hashed password I temporarily used for debugging

I'll squash all the fixups and force push this branch unless more changes are requested.

File: api/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JenySadadia) [reply api/main.py:365]: Yes, we can also optimize `/token` with that.

**Comment by gctucker** [comment]
> I reformatted new e2e tests file with a black command

Yes, feel free to use it as it's a powerful tool.  However I wouldn't recommend making it a requirement as it can cause a lot of churn, and some text editors and personal taste can induce small variants of style that are perfectly acceptable so I would rather leave the checks with just `pycodestyle`.  Also I would hope the `black` output passes `pycodestyle`.

**Comment by pawiecz** [comment]
> Also I would hope the `black` output passes `pycodestyle`.

I just confirmed that - `pycodestyle` did not raise any warnings/errors on the reformatted file.

**Review by gctucker — APPROVED** [review]

**Pull request merged**